[UI/UX:InstructorUI] Clarify poll edit confirmation

AdamTuhacek opened the pull request:
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant

### What is the current behavior?
In the edit poll screen, edit is listed as the title as well as the button to save the changes, which can be misleading.

### What is the new behavior?
Change the confirmation button to save changes.

Fixes #9090

shailpatels reviewed (approved):
Looks good, here is a screenshot of the UI changes

![image](https://user-images.githubusercontent.com/12129065/227841196-d1ae78ef-40c7-4a58-ad77-77b82c78b03f.png)

hansongu123 commented:
Personally, I like just "Save", or at least capitalize "changes" to fit with the style of other buttons.

AdamTuhacek commented:
> Personally, I like just "Save", or at least capitalize "changes" to fit with the style of other buttons.

Yeah, I corrected it to have the C capitalized. I'm not sure whether save/save changes would make much of a difference from the user perspective, I think either makes more sense than before.

hansongu123 commented:
"Save Poll", since it is "Edit Poll" before?

hansongu123 commented:
In the `polls.spec.js` file, also change the part where the automation test is invoking the button by the text on it, for example: 
![image](https://user-images.githubusercontent.com/123511202/228387652-83c2e3ed-ead7-4a07-826d-e7021d574a37.png)
to make sure it passes the test.

AdamTuhacek commented:
> In the `polls.spec.js` file, also change the part where the automation test is invoking the button by the text on it, for example: ![image](https://user-images.githubusercontent.com/123511202/228387652-83c2e3ed-ead7-4a07-826d-e7021d574a37.png) to make sure it passes the test.

Ok, thanks! I updated the name to save poll and changed the cypress test case.

hansongu123 commented:
Also on `line 289`, do the same change to pass the e2e test. Make sure you're changing it for every occurrence of the button text being checked.

hansongu123 commented on site/app/templates/polls/PollForm.twig:
```diff
@@ -163,7 +163,7 @@
         <span id="file-size-error" class="red-message polls-submit-error" hidden>Please make sure to upload a file that does not exceed the maximum size.<br/></span>
         <a href="{{base_url}}" class="btn btn-danger">Cancel</a>
-        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Edit' }} Poll</button>
+        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Save Poll' }} </button>
```
Also revert the irrelevant structural change on how this button's text is formulated. Only change the part where it says 'Edit'.

AdamTuhacek replied:
Ok, thank you very much for the detailed feedback! Changes are made

williamjallen commented on site/app/templates/polls/PollForm.twig:
```diff
@@ -163,7 +163,7 @@
         <span id="file-size-error" class="red-message polls-submit-error" hidden>Please make sure to upload a file that does not exceed the maximum size.<br/></span>
         <a href="{{base_url}}" class="btn btn-danger">Cancel</a>
-        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Edit' }} Poll</button>
+        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : : 'Save' }} Poll </button>
```
Is the extra colon a typo here?

```suggestion
        <button type="submit" id="poll-form-submit" class="btn btn-primary" {{ poll is null ? 'disabled' }}>{{ poll is null ? 'Add' : 'Save' }} Poll </button>
```

hansongu123 replied:
Looks like it

AdamTuhacek replied:
Oops, yes, thanks for catching that.

hansongu123 commented:
Sorry, I got confused when you mentioned about Polls' Cypress tests, and I just took a closer look. You shouldn't be modifying the `polls.spec.js` file because there isn't an element invoking that involves getting the content string of the changed text, instead when adding or saving changes to a poll, js would just invoke the button by looking up its type (for example in line 293). You should be good if you revert all the changes you made in the Cypress file (passes locally when I did it). Give it a local test, and don't forget to do a `submitty_install_site` to update the change. Also keep an eye on the duplicated polls named "Poll Cypress Test" if you failed to make it to the end locally and it doesn't get properly deleted.

hansongu123 commented on site/cypress/e2e/polls.spec.js:
```diff
@@ -241,7 +241,7 @@ describe('Test cases revolving around polls functionality', () => {
         cy.contains('Poll Cypress Test').siblings(':nth-child(1)').children().click();
         cy.url().should('include', 'sample/polls/editPoll');
-        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
+        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Save Poll');
```
```suggestion
        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
```

hansongu123 commented on site/cypress/e2e/polls.spec.js:
```diff
@@ -286,7 +286,7 @@ describe('Test cases revolving around polls functionality', () => {
         cy.contains('Poll Cypress Test').siblings(':nth-child(1)').children().click();
         cy.url().should('include', 'sample/polls/editPoll');
-        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
+        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Save Poll');
```
```suggestion
        cy.get('#breadcrumbs > :nth-child(7) > span').should('have.text', 'Edit Poll');
```

hansongu123 reviewed (commented):
These changes only change the check of the header in the poll editing page. It is really confusing indeed since the text is the same as in that button in `PollForm.twig`

AdamTuhacek commented:
> These changes only change the check of the header in the poll editing page. It is really confusing indeed since the text is the same as in that button in `PollForm.twig`

Awesome, thanks for all the help Hanson!
